rj_gameplay/rj_gameplay/gameplay_node.py

```diff
@@ -290,7 +312,14 @@ def add_goals_to_global_obs(self, global_obstacles, game_info):
         global_obstacles.rectangles = our_goal + their_goal
 
     def add_ball_to_global_obs(self, global_obstacles, game_info):
-        """Adds circular no-fly zone around ball during stops or restarts, to comply with rulebook."""
+        """
+        Adds circular no-fly zone around ball during stops or restarts,
+        to comply with rulebook.
+        """
+
+        # this is only ever called when self.field is filled in
+        assert self.field is not None
+
```

Comment:
I introduced some other mypy errors related to self.field being None and "None type has no attribute ball.pos", so I fixed them the dumb way by adding assert checks before each of these methods. This is fair logically because these methods are only ever called after self.field is initialized, but is probably not the clean way to go about it.